### PR TITLE
Migrage all gPRC requests to new path schema

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

A part of #3024

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: manifests/pipecd/templates/envoy-configmap.yaml [hunk 1]
```diff
@@ -61,21 +61,39 @@ data:
...
+                        prefix: /grpc.service.pipedservice.PipedService/
+                        grpc:
+                      route:
+                        cluster: grpc-piped-service
                     - match:
                         prefix: /pipe.api.service.pipedservice.PipedService/
                         grpc:
                       route:
                         cluster: grpc-piped-service
+                        prefix_rewrite: /grpc.service.pipedservice.PipedService/
```
Comment (nghialv) [inline manifests/pipecd/templates/envoy-configmap.yaml:74]: This allows the new control-plane can be able to accept gRPC requests from both old and new Piped agents.

**Comment by nghialv** [comment]
Hold untile `v0.24.0` got released.

**Comment by nakabonne** [comment]
What's going on?
/trigger test

**Comment by khanhtc1202** [comment]
> What's going on?

```
//pkg/app/web:test                                                      TIMEOUT in 60.1s
```
web-test's prob 😢

**Comment by khanhtc1202** [comment]
Nice 👍 
/approve

**Pull request merged**